src/Centreon/Domain/Entity/EntityValidator.php

```diff
@@ -233,14 +234,20 @@ private function getConstraints(string $entityName, array $groups, bool $firstCa
     private function removeDuplicatedViolation(
         ConstraintViolationListInterface $violations
     ): ConstraintViolationListInterface {
-        $violationCodes = [];
         /**
          * @var $violation ConstraintViolationInterface
          */
-        for ($index = 0; $index < count($violations); $index++) {
+        $violationCodes = [];
+        $violationNumber = count($violations);
+        for ($index = 0; $index < $violationNumber; $index++) {
```

Comment:
could we use foreach ? (to avoid count violations on the previous line

Reply:
No, because I need the index to be able to remove a violation using their index